doc/whats_new/upcoming_changes/sklearn.utils/30775.fix.rst

```diff
@@ -0,0 +1,7 @@
+-  :func:`utils.estimator_checks._enforce_estimator_tags_y` enforces a binary `y`
```

Comment:
Let's not reference the private API in the changelog. We just need to mention that the 2 checks have been for binary targets.